### ISSUE TITLE
Maccy takes several minutes to start after upgrade to MacOS 13.4.1

### ISSUE THREAD

**Issue by antivirtel** [p0deje/Maccy#574]
Hi,

I've just upgraded MacOS `13.4` -> `13.4.1 (22F82)`, after the upgrade I also upgraded all brew packages, incl. Maccy `0.26.4` -> `0.26.5`, didn't notice anything right away. However, after booted today, Maccy didn't start. Tried killing it, starting from Terminal, no output or error. I see the process running:

```
$ ps aux | grep -i maccy
user             6191   0.0  0.0 33588380    124 s001  R+   12:16am   0:00.00 grep --color -i maccy
user             6069   0.0  0.6 34542052 106012   ??  S    12:12am   0:02.58 /Applications/Maccy.app/Contents/MacOS/Maccy
```

I've tried downgrading to `0.26.4` too (uninstall from brew, install via [old brew Ruby file](https://github.com/Homebrew/homebrew-cask/blob/c22204f657789215cb5d3e411255109b1a76249c/Casks/maccy.rb)), same thing, doesn't work.

I've also tried to remove it from Settings -> Privacy & Security -> Accessibility, then re-adding, no change.

I'm not too familiar with MacOS debugging (only Linux), is there a log file to see, some way of telling why doesn't it start?

**Update**: The older version of `0.26.4` eventually started. Will try upgrading Maccy again tomorrow...

Thanks!

**Comment by p0deje** [comment]
1. Can you please share the output of running `defaults read org.p0deje.Maccy` in Terminal.app?
2. Can you open Console.app, search for "Process: Maccy", hit "Start", start Maccy and share the output from the Console.app.

**Comment by antivirtel** [comment]
Hi @p0deje,

Thanks for coming back this quickly. The requested outputs:

### 1. Config
Dumped from `0.26.5`:
<details><summary>0.26.5 config</summary>
<p>

```
{
    "KeyboardShortcuts_popup" = "{\\"carbonModifiers\\":768,\\"carbonKeyCode\\":8}";
    "LaunchAtLogin__hasMigrated" = 1;
    "NSWindow Frame SUStatusFrame" = "760 691 400 134 0 0 1920 1055 ";
    "NSWindow Frame SUUpdateAlert" = "530 484 620 398 0 0 1680 1025 ";
    "NSWindow Frame com.sindresorhus.Preferences.FrameAutosaveName" = "1023 899 542 440 0 0 2560 1415 ";
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2023-06-12 15:42:29 +0000";
    SUUpdateGroupIdentifier = 1238859504;
    WebKitDefaultFontSize = 13;
    WebKitJavaScriptEnabled = 0;
    WebKitPluginsEnabled = 0;
    WebKitStandardFont = "-apple-system-font";
    WebKitUserStyleSheetEnabledPreferenceKey = 1;
    WebKitUserStyleSheetLocationPreferenceKey = "/Applications/Maccy.app/Contents/Frameworks/Sparkle.framework/Resources/ReleaseNotesColorStyle.css";
    enabledPasteboardTypes =     (
        "public.utf8-plain-text",
        "public.png",
        "public.file-url",
        "public.html",
        "public.rtf",
        "public.tiff"
    );
    history = {length = 42, bytes = 0x62706c69 73743030 a0080000 00000000 ... 00000000 00000009 };
    historySize = 999;
    ignoredPasteboardTypes =     (
        "com.agilebits.onepassword",
        "net.antelle.keeweb",
        "Pasteboard generator type",
        "de.petermaurer.TransientPasteboardType",
        "com.typeit4me.clipping"
    );
    migrations =     {
        "2020-02-22-history-item-add-copied-at" = 1;
        "2020-02-22-history-item-add-number-of-copies" = 1;
        "2020-04-18-switch-storage-to-core-data" = 1;
        "2020-04-25-allow-custom-ignored-types" = 1;
        "2020-06-19-use-keyboardshortcuts" = 1;
        "2020-09-01-ignore-keeweb" = 1;
        "2021-02-20-allow-to-customize-supported-types" = 1;
        "2021-06-28-add-title-to-history-item" = 1;
        "2021-10-16-remove-dynamic-pasteboard-types" = 1;
        "2022-08-01-rename-suppress-clear-alert" = 1;
        "2022-11-14-add-html-rtf-to-supported-types" = 1;
        "2023-01-22-add-regexp-search-mode" = 1;
    };
    pasteByDefault = 1;
    popupPosition = cursor;
    searchMode = exact;
    sortBy = lastCopiedAt;
}
```

</p>
</details> 


### 2. Logs
Launch of `0.26.4` at 12:46:07, the app was ready at 12:53:27. So it took 7 minutes and 20 seconds to start. Log:
<details><summary>0.26.4 launch log</summary>
<p>

```
default	12:46:07.067246+0100	Maccy	Requesting container lookup; personaid = -1, type = NOPERSONA, name = <unknown>, origin [pid = 0, personaid = 0], proximate [pid = 0, personaid = 0], class = 2, identifier = <private>, group_identifier = <private>, create = 0, temp = 0, euid = 502, uid = 502
default	12:46:07.067303+0100	Maccy	Query; personaid = -1, type = NOPERSONA, name = <unknown>, origin [pid = 0, personaid = 0], proximate [pid = 0, personaid = 0], euid = 502, uid = 502, query = <private>
default	12:46:07.069645+0100	Maccy	container_query_get_single_result: success
default	12:46:07.069717+0100	Maccy	Consuming sandbox extension; path = [<private>], key = 0
default	12:46:07.069787+0100	Maccy	Consuming sandbox extension; path = [<private>], key = 0
default	12:46:07.069849+0100	Maccy	Revoking sandbox extension; key = 0
default	12:46:07.069935+0100	Maccy	container_create_or_lookup_for_platform: success
default	12:46:07.069984+0100	Maccy	Revoking sandbox extension; key = 0
default	12:46:07.117331+0100	Maccy	server port 0x00003207, session port 0x00003207
default	12:46:07.149460+0100	Maccy	Received configuration update from daemon (initial)
default	12:46:07.150933+0100	Maccy	New connection 0x7f1e7 main
default	12:46:07.155034+0100	Maccy	CHECKIN: pid=2542
default	12:46:07.162951+0100	Maccy	CHECKEDIN: pid=2542 asn=0x0-0x75075 foreground=0
default	12:46:07.164669+0100	Maccy	FRONTLOGGING: version 1
default	12:46:07.164714+0100	Maccy	Registered, pid=2542 ASN=0x0,0x75075
default	12:46:07.166372+0100	Maccy	BringForward: pid=2542 asn=0x0-0x75075 bringForward=0 foreground=0 uiElement=1 launchedByLS=1 modifiersCount=1 allDisabled=0
default	12:46:07.166409+0100	Maccy	BringFrontModifier: pid=2542 asn=0x0-0x75075 Modifier 0 hideAfter=0 hideOthers=0 dontMakeFrontmost=0 mouseDown=0/0 seed=0/0
default	12:46:07.167861+0100	Maccy	Current system appearance, (HLTB: 2), (SLS: 1)
default	12:46:07.170042+0100	Maccy	No persisted cache on this platform.
default	12:46:07.170712+0100	Maccy	Failed to copy the SysCfgDict MG key with error: 0
default	12:46:07.171833+0100	Maccy	Current system appearance, (HLTB: 2), (SLS: 1)
default	12:46:07.172542+0100	Maccy	Post-registration system appearance: (HLTB: 2)
default	12:46:07.194180+0100	Maccy	Registering for test daemon availability notify post.
default	12:46:07.194345+0100	Maccy	notify_get_state check indicated test daemon not ready.
default	12:46:07.194458+0100	Maccy	notify_get_state check indicated test daemon not ready.
default	12:46:07.194561+0100	Maccy	notify_get_state check indicated test daemon not ready.
default	12:46:07.200159+0100	Maccy	Received DFR status change 0. status = 0x1
default	12:46:07.200822+0100	Maccy	Received DFR status change 1. status = 0x1
default	12:46:07.202066+0100	Maccy	SignalReady: pid=2542 asn=0x0-0x75075
default	12:46:07.202529+0100	Maccy	SIGNAL: pid=2542 asn=0x0x-0x75075
default	12:46:07.217174+0100	Maccy	Reacting to DFR status change 0. status = 0x1
default	12:46:07.217648+0100	Maccy	Reacting to DFR status change 1. status = 0x1
default	12:46:07.234774+0100	Maccy	NSApp cache appearance:
-NSRequiresAquaSystemAppearance: 0
-appearance: (null)
-effectiveAppearance: <NSCompositeAppearance: 0x600000a04680
 (
    "<NSDarkAquaAppearance: 0x600000a04900>",
    "<NSSystemAppearance: 0x600000a04400>"
)>
default	12:46:07.242763+0100	Maccy	order window: b1 op: 1 relative: b1 related: 0
default	12:46:17.178108+0100	Maccy	LSExceptions shared instance invalidated for timeout.
default	12:53:27.956643+0100	Maccy	NotifyToken::RegisterDispatch(user.uid.502.com.apple.LaunchServices.database) fired for session key <private>
default	12:53:41.127420+0100	Maccy	SetFrontProcess: asn=0x0-0x75075 options=1
default	12:53:41.140914+0100	Maccy	Shortcuts Contextual Actions enabled
default	12:53:41.532325+0100	Maccy	order window: e6 op: 1 relative: 0 related: 0
default	12:53:43.999814+0100	Maccy	order window: e6 op: 0 relative: 0 related: 0
default	12:53:43.004317+0100	Maccy	order window: e6 op: 0 relative: 0 related: 0
```

</p>
</details>

Launch of `0.26.5` at 13:11:32, start of the app at 13:15:38. So it took 4 minutes and 6 seconds to start. Log:

<details><summary>0.26.5 launch log</summary>
<p>

```
default	13:11:32.675062+0100	Maccy	Requesting container lookup; personaid = -1, type = NOPERSONA, name = <unknown>, origin [pid = 0, personaid = 0], proximate [pid = 0, personaid = 0], class = 2, identifier = <private>, group_identifier = <private>, create = 0, temp = 0, euid = 502, uid = 502
default	13:11:32.675132+0100	Maccy	Query; personaid = -1, type = NOPERSONA, name = <unknown>, origin [pid = 0, personaid = 0], proximate [pid = 0, personaid = 0], euid = 502, uid = 502, query = <private>
default	13:11:32.678103+0100	Maccy	container_query_get_single_result: success
default	13:11:32.678174+0100	Maccy	Consuming sandbox extension; path = [<private>], key = 0
default	13:11:32.678231+0100	Maccy	Consuming sandbox extension; path = [<private>], key = 0
default	13:11:32.678246+0100	Maccy	Revoking sandbox extension; key = 0
default	13:11:32.678257+0100	Maccy	container_create_or_lookup_for_platform: success
default	13:11:32.678271+0100	Maccy	Revoking sandbox extension; key = 0
default	13:11:32.733450+0100	Maccy	server port 0x00003207, session port 0x00003207
default	13:11:32.769836+0100	Maccy	Received configuration update from daemon (initial)
default	13:11:32.771362+0100	Maccy	New connection 0x70b2b main
default	13:11:32.775629+0100	Maccy	CHECKIN: pid=5374
default	13:11:32.784795+0100	Maccy	CHECKEDIN: pid=5374 asn=0x0-0xa90a9 foreground=0
default	13:11:32.789097+0100	Maccy	FRONTLOGGING: version 1
default	13:11:32.789181+0100	Maccy	Registered, pid=5374 ASN=0x0,0xa90a9
default	13:11:32.791101+0100	Maccy	BringForward: pid=5374 asn=0x0-0xa90a9 bringForward=0 foreground=0 uiElement=1 launchedByLS=1 modifiersCount=1 allDisabled=0
default	13:11:32.791167+0100	Maccy	BringFrontModifier: pid=5374 asn=0x0-0xa90a9 Modifier 0 hideAfter=0 hideOthers=0 dontMakeFrontmost=0 mouseDown=0/0 seed=0/0
default	13:11:32.792223+0100	Maccy	Current system appearance, (HLTB: 2), (SLS: 1)
default	13:11:32.796078+0100	Maccy	No persisted cache on this platform.
default	13:11:32.796773+0100	Maccy	Failed to copy the SysCfgDict MG key with error: 0
default	13:11:32.797944+0100	Maccy	Current system appearance, (HLTB: 2), (SLS: 1)
default	13:11:32.799633+0100	Maccy	Post-registration system appearance: (HLTB: 2)
default	13:11:32.832376+0100	Maccy	Registering for test daemon availability notify post.
default	13:11:32.832628+0100	Maccy	notify_get_state check indicated test daemon not ready.
default	13:11:32.832818+0100	Maccy	notify_get_state check indicated test daemon not ready.
default	13:11:32.832964+0100	Maccy	notify_get_state check indicated test daemon not ready.
default	13:11:32.846396+0100	Maccy	Received DFR status change 0. status = 0x1
default	13:11:32.847918+0100	Maccy	Received DFR status change 1. status = 0x1
default	13:11:32.852401+0100	Maccy	SignalReady: pid=5374 asn=0x0-0xa90a9
default	13:11:32.853229+0100	Maccy	SIGNAL: pid=5374 asn=0x0x-0xa90a9
default	13:11:32.882519+0100	Maccy	Reacting to DFR status change 0. status = 0x1
default	13:11:32.883025+0100	Maccy	Reacting to DFR status change 1. status = 0x1
default	13:11:32.913877+0100	Maccy	NSApp cache appearance:
-NSRequiresAquaSystemAppearance: 0
-appearance: (null)
-effectiveAppearance: <NSCompositeAppearance: 0x600003a8e780
 (
    "<NSDarkAquaAppearance: 0x600003a8f200>",
    "<NSSystemAppearance: 0x600003a90400>"
)>
default	13:11:32.926816+0100	Maccy	order window: 127 op: 1 relative: 127 related: 0
default	13:11:42.166404+0100	Maccy	NotifyToken::RegisterDispatch(user.uid.502.com.apple.LaunchServices.database) fired for session key <private>
default	13:11:42.807080+0100	Maccy	LSExceptions shared instance invalidated for timeout.
default	13:13:19.728526+0100	Maccy	Received DFR status change 2. status = 0x5
default	13:13:39.367885+0100	Maccy	Received DFR status change 3. status = 0x1
default	13:14:34.755781+0100	Maccy	Reacting to DFR status change 2. status = 0x5
default	13:14:34.755860+0100	Maccy	Reacting to DFR status change 3. status = 0x1
default	13:15:38.186064+0100	Maccy	NotifyToken::RegisterDispatch(user.uid.502.com.apple.LaunchServices.database) fired for session key <private>
default	13:15:48.237874+0100	Maccy	NotifyToken::RegisterDispatch(user.uid.502.com.apple.LaunchServices.database) fired for session key <private>
default	13:15:48.528042+0100	Maccy	SetFrontProcess: asn=0x0-0xa90a9 options=1
default	13:15:48.541239+0100	Maccy	Shortcuts Contextual Actions enabled
default	13:15:49.052289+0100	Maccy	order window: 144 op: 1 relative: 0 related: 0
default	13:15:55.967705+0100	Maccy	order window: 144 op: 0 relative: 0 related: 0
default	13:15:55.968450+0100	Maccy	order window: 144 op: 0 relative: 0 related: 0
default	13:15:55.974829+0100	Maccy	order window: 144 op: 0 relative: 0 related: 0
default	13:15:55.975419+0100	Maccy	order window: 144 op: 0 relative: 0 related: 0
```

</p>
</details> 


Thank you!

**Comment by p0deje** [comment]
@antivirtel Could be that you copied something very big and it takes a lot of time to load it by Maccy. Can you try clearing the history?

**Comment by antivirtel** [comment]
I'd rather try to identify it, and only delete that one. Is there a way to check the size of the pastes in the DB? Where is the DB located? I assume it's just an SQLite file, isn't it? Thanks

**Comment by p0deje** [comment]
The database is in $HOME/Library/Containers/org.p0deje.Maccy/Data/Library/Application Support/Maccy/, you can check what's inside.

**Comment by antivirtel** [comment]
Ok, I've checked, is 304 KiB too big for a paste?
<details><summary>Top 20 pastes' sizes</summary>
<p>

```SQL
SELECT Z_PK, length(ZVALUE)/1024 FROM ZHISTORYITEMCONTENT ORDER BY length(ZVALUE) DESC LIMIT 20;
```

```
Z_PK	length(ZVALUE)/1024
27451	304
24940	130
24949	130
24964	130
24971	130
27223	87
26908	66
26921	66
26948	59
27559	42
27569	42
26757	42
26767	42
25186	41
25187	41
27267	41
26428	41
26363	41
26429	41
27266	41

```

</p>
</details>

Can remove the ones above 100 KiB, but the whole file is only 8 MiB:
```
~/Library/Containers/org.p0deje.Maccy/Data/Library/Application Support/Maccy$ ls -lh
total 9.0M
-rw-r--r-- 1 user group 8.3M Jul  6 16:27 Storage.sqlite
```

**Comment by p0deje** [comment]
Doesn't look too big, maybe there is something in the contents that makes it load slower than usual? Can you try running 0.25 or 0.24 and see if it helps?

**Comment by antivirtel** [comment]
After a whole day's work, the top 20 biggest has changed and since it can only take 999 entries, the problematic one(s) are now gone or something else changed, and now it starts promptly with the latest version, and the last 2 versions you mentioned too. The 304 KiB paste is still in the DB:

<details><summary>Top 20 after work</summary>
<p>

```SQL
SELECT Z_PK, length(ZVALUE)/1024 FROM ZHISTORYITEMCONTENT ORDER BY length(ZVALUE) DESC LIMIT 20;
```

```
Z_PK	length(ZVALUE)/1024
27451	304
27223	87
26908	66
26921	66
26948	59
27559	42
27569	42
26757	42
26767	42
25186	41
25187	41
27267	41
26428	41
26363	41
26429	41
27266	41
26361	41
26432	41
25933	40
25939	40
```

</p>
</details> 

I've backed the DB up in the morning before opening it up in an SQLite client, so the problematic paste(s) meant to be in it after I moved the file back. I've tried all the 3 versions, all starts promptly now. 🤷🏼 I'll try to see if a cold boot causes it maybe (tomorrow). 

Ps. I have CrowdStrike Falcon-Sensor running, which is a black box, downloads company policy, can't stop it even with admin rights on the Mac, that can do literally anything it wants, but since I started using Maccy (2022 Mar 10) never had an issue (till this week).

Will let you know tomorrow, thanks!
